processing/src/main/java/io/druid/query/filter/SelectorDimFilter.java

```diff
@@ -225,4 +255,18 @@ private void initLongValue()
       longsInitialized = true;
```

Comment:
If caching resulting predicate directly instead of `valueAsLong` (it's possible because predicates are immutable), `longsInitialized` is not needed. Also, locking could be removed and benign race allowed.